frontend/src/my-books/MyBooks.tsx

```diff
@@ -15,23 +15,85 @@ You should have received a copy of the GNU General Public License along with thi
 If not, see <https://www.gnu.org/licenses/>.
 */
 
-import React, { Component } from 'react'
+
+import React, { Component, ReactElement } from 'react'
+import { NavBar } from '../shared/navigation/NavBar'
+
 import { ShelfCarousel } from '../shared/book-display/ShelfCarousel'
 import { Layout } from "../shared/components/Layout";
 import './MyBooks.css'
+import Button from '@material-ui/core/Button';
+import ShelfModal from './myBookModal/ShelfModal';
+import Hidden from "@material-ui/core/Hidden";
+import MenuIcon from '@material-ui/icons/Menu';
 
 interface IState {
+    showShelfModal: boolean,
 }
 
 class MyBooks extends Component<Record<string, unknown>, IState> {
-    render() {
+    constructor(props: Record<string, unknown>) {
+        super(props)
+        this.state = {
+        showShelfModal: false,
+        }
+        this.onAddShelf = this.onAddShelf.bind(this)
+        this.onAddShelfModalClose = this.onAddShelfModalClose.bind(this)
+    }
+
+    onAddShelf(): void {
+        this.setState({
+            showShelfModal: true
+        })
+    }
+    
+    onAddShelfModalClose(): void {
+        this.setState({
+            showShelfModal: false,
+        })
+    }
+    render(): ReactElement {
         return (
-            <Layout title="My books">
-                <ShelfCarousel title="Reading" />
-                <ShelfCarousel title="To Read" />
-                <ShelfCarousel title="Read" />
-                <ShelfCarousel title="Did not finish" />
-            </Layout>
+
+            <div>
+                <NavBar />
+
+                
+                <Hidden smDown implementation="css">
+                <div className="my-book-top">
+                    <h1>My books</h1>
+                    <div className="my-book-top-buttons">
+                    <Button
+                    variant="contained"
+                    color="primary">
+                        Add Book
+                    </Button>
+                    <Button
+                     onClick={this.onAddShelf}
+                    variant="contained"
+                    color="primary">
+                        Add Shelf
+                    </Button>
+                    </div>  
+                    </div> 
+                </Hidden>
+
+                <Hidden mdUp implementation="css">
+                    <MenuIcon className="my-book-top" />                              
+                </Hidden>
+         
+                <div>
+                    <ShelfCarousel title="Reading" />
+                    <ShelfCarousel title="To Read" />
+                    <ShelfCarousel title="Read" />
+                    <ShelfCarousel title="Did not finish" />
+                </div>
+                <ShelfModal 
+                    open={this.state.showShelfModal} 
+                    onClose={this.onAddShelfModalClose}
+                />
+            </div>
+
```

Comment:
Here, line 63 is a child of line 62, so we can indent line 63 accordingly and edit indentation for this entire block, No whitespace is needed for lines 80, 84, 96.